src/hotspot/share/runtime/os.hpp

```diff
@@ -534,6 +534,7 @@ class os: AllStatic {
   static void   realign_memory(char *addr, size_t bytes, size_t alignment_hint);
 
   // NUMA-specific interface
+  static bool   numa_set_thread_affinity(Thread* thread, int node);
```

Comment:
How to interpret the returned value? What do you suggest callers do with the return-value?

Reply:
NUMA affinity is a performance hint, not a requirement. Callers can’t meaningfully react to failure -- retry is unlikely to succeed, and there’s no feasible fallback strategy in most contexts.

> since the algorithm and functionality works if setting the affinity would fail

I can't think of a scenario where numa-affinity affects correctness.

Ofc, this is very subjective. Up to you.

Reply:
Such logic (logging + skip) would complicate the consumer flow with little gain -- failure in this (kind of) API should be no-op (as if skipped).

Also not-try-again assumes future calls always fail, but I'm not sure if that holds. Logging a failure might be interesting, but it should probably be done at OS level, as we use `false` to mean either failure or unsupported.